### ISSUE TITLE
Duplicate handlers can be specified when updating proportions

### ISSUE THREAD

**Issue by mgcolburn** [dforce-network/dToken#14]
Severity: Low

### Description 
For example, imagine there are four handlers each with weight `250000`. Now the proportions are updated with handlers `[handler1, handler1, handler1, handler1]` and values `[1000000,0,0,0]`

```
 function setHandlers( 
     address[] memory _handlers, 
     uint256[] memory _proportions 
 ) private { 
     require( 
         _handlers.length == _proportions.length && _handlers.length > 0, 
         "setHandlers: handlers & proportions should not have 0 or different lengths" 
     ); 
  
     // The 1st will act as the default handler. 
     defaultHandler = _handlers[0]; 
  
     uint256 _sum = 0; 
     for (uint256 i = 0; i < _handlers.length; i++) { 
         require( 
             _handlers[i] != address(0), 
             "setHandlers: handlerAddr contract address invalid" 
         ); 
  
         _sum = _sum.add(_proportions[i]); 
  
         handlers.push(_handlers[i]); 
         proportions[_handlers[i]] = _proportions[i]; 
         isHandlerActive[_handlers[i]] = true; 
     } 
  
     // The sum of proportions should be 1000000. 
     require( 
         _sum == totalProportion, 
         "the sum of proportions must be 1000000" 
     ); 
 } 
```

This will cause `handler1` to have a weight of `0` and the system to have a total weight of only `750000` (a similar scenario could result in a weight sum that is greater than `1000000`). However, the sanity check for the sum at the end of the function will not revert as the sum it calculates is across the `_proportions` input array, not the actual state of the `proportions` mapping at the end of the update.

### Recommendation
Add an explicit check to `updateProportions` to check for duplicate Handlers.

**Comment by Donald-Nobel** [comment]
Known issue, we will fix it.

**Comment by bamboo-seven** [comment]
Fixed in 65116fa42b0a8b1d43182372fc35031026d22f45
